examples/tool_tests/gowin.c

```diff
@@ -1,10 +1,13 @@
-#pragma PART "GW2AR-18C GW2AR-LV18QN88PC8/I7"
+// @tiagosr: adding --device_version (or --device_name) at the end as 
+// PipelineC currently recognizes vendors by the first letters of the part name
+#pragma PART "GW2AR-LV18QN88PC8/I7 --device_version C"
```

Comment:
So the pipelinec tool should start cacheing some synthesis results under
`path_delay_cache/gowin/pnr/<PART>/BIN_OP_PLUS_uint16_t_uint16_t.delay` for example

What I am worried about is how the gowin part string as provided as spaces and could make for ugly directory name
`path_delay_cache/gowin/pnr/GW2AR-LV18QN88PC8/I7 --device_version C/BIN_OP_PLUS_uint16_t_uint16_t.delay`
hmmm

Reply:
I see - as I saw it in the tool, `--name` and `--device_version` are a weird kind of alias, where `--name` has the same info that the part number already has, except for the device version. And I have no idea why they have both there. Personally I could go either way, although `--device_version` seems slightly less redundant.

Reply:
^ related to the `pnr` in above

Is this flow that produces timing report based on full place and route final timing numbers or pre-route synthesis estimates?

Reply:
Maybe a hash/CRC of the #pragma PART string could replace the name, as the directory?

Reply:
If you look at https://github.com/JulianKemmerer/PipelineC/blob/5c8606c51254ce7376ce4db7dbd56da1ac6f040e/src/EFINITY.py#L18

the Efinity tool also has a 'part + other redundant info' kind of situation
and what I did there was just only provide the part (no spaces etc) and then look up the other flags (family, timing model) based on part for the user... might work here too

Reply:
And ooof the `/I7` slash in the part name - is that needed - also maybe can lookup? or heck just replace char before using it as dir name hmm

Reply:
On the flow that generates the timing report, yes, it bases numbers on full pnr final timings. The `run all` line in the Tcl script runs the 2 phases the system offers: `run syn` for synthesis and `run pnr`, which relies on info from `run syn`

Reply:
Is there any option in gowin tools to get pre route syn timing estimates? (can be faster to iterate with than full pnr)

Reply:
Regarding hash - certainly could work

but I like human readability of the path delay cache dir name - otherwise maybe would consider some binary file database or something maybe instead of files? 
also there are some folks who like to try and model specific parts, ex. look up all the adders for xilinx 7 series part by name etc - and its easy if dir name is part name, otherwise need to tell them to hash the name to do lookup etc  (unless just saying only for gowin hash the part...but still hmmm)
little more complicated than id prefer hmmm

Reply:
On pre-route syn timing estimates, I think we might have something... Though it will include scraping HTML, which the synthesis reports are emitted to, with no option to output a text file. I mean, might be less complicated than expected.
[my_pipeline_0CLK_b45f1687_top_syn.rpt.html.txt](https://github.com/JulianKemmerer/PipelineC/files/14152580/my_pipeline_0CLK_b45f1687_top_syn.rpt.html.txt)

Reply:
Oh neat, its very nice to have those quick syn estimates

yeah parsing html if we have to :shrug:  - imo wouldnt jump to doing that right away before getting other stuff working

Reply:
So, I added some data to deconstruct a Gowin part and be able to select the device grades (and as a bonus, give options on device versions). As their grade selection is not a trivial grid of commercial/industrial equivalents, the logic became slightly more complicated than expected.